src/python/zquantum/core/estimator.py

```diff
@@ -129,17 +152,46 @@ def get_estimated_expectation_values(
             frame_circuits.append(circuit + frame_circuit)
             frame_operators.append(frame_operator)
 
-        if n_samples is not None:
-            logger.warning(
-                f"""Using n_samples={n_samples} (argument passed to get_estimated_expectation_values). 
-                    Ignoring backend.n_samples={backend.n_samples}"""
+        if shot_allocation_strategy == "uniform":
+
+            if n_total_samples is not None:
+                raise ValueError(
+                    "Uniform sampling does not yet support n_total_samples."
+                )
+
+            if n_samples is not None:
+                logger.warning(
+                    f"""Using n_samples={n_samples} (argument passed to get_estimated_expectation_values). 
+                        Ignoring backend.n_samples={backend.n_samples}"""
+                )
+                n_samples = (n_samples,) * len(frame_circuits)
+                measurements_set = backend.run_circuitset_and_measure(
+                    frame_circuits, n_samples
+                )
+            else:
+                measurements_set = backend.run_circuitset_and_measure(frame_circuits)
+
+        elif shot_allocation_strategy == "optimal":
```

Comment:
It looks to me that since we have two very different cases, with two different input params, maybe these should be two different classes?

Reply:
I combined some of the duplicated code across the different cases in cf9fdb6245461d9cfaca77dfc879a55905e9c315. If/when we add support for `n_total_samples` to the `shot_allocation_strategy="uniform"` case, then the two cases will be more similar.

Splitting it into two classes though wouldn't address the remaining edge case (that `n_samples` can't be used with `shot_allocation_strategy="optimal"`) because you'd still inherit `n_samples` from the base `Estimator` class. Perhaps we should change `Estimator` to have `n_total_samples` instead of `n_sampels`? Although this feels like a story for another day.

Reply:
👍